### DISCUSSION TITLE
How to create infra with previously created manifest.yml file

### DISCUSSION THREAD

**Discussion by gauravjeswani** [aws/copilot-cli#2899] [Q&A] [answered]
I initialized aws copilot application, so it created a copilot folder and inside that it created service details with manifest.yml file. And created my AWS infrastructure for me. After that I delete that infra using **_copilot app delete_**.

Now I want to deploy this without going to initialization process of copilot. How is that possible as manifest.yml is already having those details?

Or in another words how I can replicate same application in another AWS account?

**Comment by Lou1415926** [comment, marked as answer]
Hello @gauravjeswani ! You want to deploy services with the same configurations as before, am I correct?

You can still run `copilot app init` to initialize an application as you did previously. Then, run `copilot svc init`. When asked for the service's name, put the same name as you have before - this information can be found in manifest's `name` field.

Copilot will check if the `copilot/<service>/manifest.yml` already exists, and won't overwrite it if it has. As a result, your previous configurations in the manifest will stay.

Next time when you run `copilot svc deploy`, Copilot will deploy the service using configurations in the manifest. Since it's the same manifest as before, the configuration will be the the same.

**Reply by gauravjeswani** [reply]
Hello @Lou1415926 thanks for you reploy, I tried running command in below order:
`copilot app init`
`copilot svc init`

And when it asked for service name I entered exactly same name which was used previously (mentioned in previous  manifest's name field), but even after that it asked for configurations again (like DockerImage, Port etc).

**Reply by Lou1415926** [reply]
Yes Copilot does try to collect the information, but the manifest won't get overwritten. This is an enhancement we can make - if the manifest already exists, we can skip asking for information, or only ask for those that are missing.

**Reply by gauravjeswani** [reply]
Yes, I worked that way. I deployed it successfully. Thanks for your help. And that totally make sense to avoid the confusion to make above suggested enhancement. Or the option to user to use existing configuration (and skip asking) or re-write it.

Thanks for your help.

**Reply by huanjani** [reply]
Hi @gauravjeswani! 
Partly in response to your feedback, we have made the UX for your use case more smooth: https://github.com/aws/copilot-cli/pull/2966. The change will be included in our upcoming release.
Thank you for your question!

**Reply by gauravjeswani** [reply]
Hello @huanjani ,

That's great. Thanks for quick resolution. Would be really helpful in distribution and re-deployment of application in different accounts.

Great Job.
